gateway-js/src/index.ts

```diff
@@ -358,18 +357,23 @@ export class ApolloGateway implements GraphQLService {
   }
 
   protected async updateComposition(): Promise<void> {
-    let result: Await<ReturnType<Experimental_UpdateServiceDefinitions>>;
-    this.logger.debug('Checking service definitions...');
-    try {
-      result = await this.updateServiceDefinitions(this.config);
-    } catch (e) {
-      this.logger.error(
-        'Error checking for changes to service definitions: ' +
-          ((e && e.message) || e),
+    this.logger.debug('Checking for composition updates...');
+
+    // This may throw, but an error here is caught and logged upstream
+    const result = await this.updateServiceDefinitions(this.config);
+
+    if (isCsdlUpdate(result)) {
+      await this.updateCsdl(result);
+    } else if (isServiceDefinitionUpdate(result)) {
+      await this.updateServiceDefs(result);
+    } else {
+      throw new Error(
+        'Programming error: unexpected result type from `updateServiceDefinitions`',
       );
-      throw e;
     }
+  }
 
+  private async updateServiceDefs(result: ServiceDefinitionUpdate): Promise<void> {
```

Comment:
This method name is confusingly close to updateServiceDefinitions.

Reply:
Agree, this name shouldn't have even made it to review.

Let me know if a3a6b81 invokes any good or bad feelings. The newly named `updateSchema` does just that in both branches. `updateComposition` I feel is now more correctly specific, since that's the only branch which is doing any composition (maybe `recomposeServiceDefinitions` instead?).